compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -173,6 +176,81 @@ pub struct NodeInterner {
     pub(crate) type_ref_locations: Vec<(Type, Location)>,
 }
 
+// TODO: relocate
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum ArithId {
+    Dummy,
+    Incremental(usize),
+}
+
+// TODO: relocate
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub enum ArithExpr {
+    Op {
+        kind: ArithOpKind,
+        lhs: Box<ArithExpr>,
+        rhs: Box<ArithExpr>,
+    },
+    Variable(Rc<String>),
```

Comment:
Should probably keep at least the type variable id if not the whole type variable here eventually, right? To differentiate identical names if needed.